node_modules/cylc-action-utils.js

```diff
@@ -53,3 +52,15 @@ exports.execSync = (cmd, options = {}) => {
     }
     return stdout;
 };
+
+exports.setEnv = (name, value) => {
+    // Set an environment variable for later steps in a workflow to use
+    const cmd = `echo "${name}=${value}" >> $GITHUB_ENV`;
+    execSync(cmd, execSyncOpts);
+};
+
+exports.addPath = (path) => {
+    // Add a path to $PATH
+    const cmd = `echo "${path}" >> $GITHUB_PATH`;
+    execSync(cmd, execSyncOpts);
+};
```

Comment:
Yesterday while reviewing @datamel 's PR https://github.com/cylc/cylc-ui/pull/517, I found a warning that' related to an CVE in GH actions toolkit (which I had never heard before). Looks like they have the tools to safely manipulate `$PATH`, and set environment variables too. Probably a good improvement for later? 

https://github.com/actions/toolkit/tree/main/packages/core#exporting-variables

Reply:
I had looked into that a few months ago, however one drawback was [the need to commit the npm dependencies](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github):
> GitHub downloads each action run in a workflow during runtime and executes it as a complete package of code before you can use workflow commands like run to interact with the runner machine. This means you must include any package dependencies required to run the JavaScript code. **You'll need to check in the toolkit `core` and `github` packages to your action's repository.**

However this isn't such a drawback now the actions are in a separate repo, admittedly. Still, I think the flexibility of these "composite run step" actions is useful compared to pure javascript actions.